noisy-simulator/src/tests.rs

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::{
+    instrument::Instrument,
+    operation::{operation, Operation},
+    NoisySimulator, TOLERANCE,
+};
+use num_complex::Complex;
+
+/// Assert that two f64 are equal up to a `TOLERANCE`.
+pub fn assert_approx_eq(left: f64, right: f64) {
+    if (left - right).abs() > TOLERANCE {
+        panic!("aprox_equal failed, left = {left}, right = {right}");
+    }
+}
+
+/// Returns an H gate.
+fn h() -> Operation {
```

Comment:
It seems like the operations defined for these tests are modelling a noisless system. It would be good to add a comment or convey that in the name of the operations because it could be unexpected that the test cases for a noisy simulator use noiseless operations.

Reply:
Yes, you are right, I changed those to `noiseless_h`, etc.